src/shims/unix/foreign_items.rs

```diff
@@ -267,6 +269,73 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(result, dest)?;
             }
 
+            "sched_getaffinity" => {
+                // Currently this function does not exist on all Unixes, e.g. on macOS.
+                if !matches!(&*this.tcx.sess.target.os, "linux" | "freebsd" | "android") {
+                    throw_unsup_format!(
+                        "`sched_getaffinity` is not supported on {}",
+                        this.tcx.sess.target.os
+                    );
+                }
+
+                let [pid, cpusetsize, mask] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let pid = this.read_scalar(pid)?.to_u32()?;
+                let cpusetsize = this.read_target_usize(cpusetsize)?;
+                let mask = this.read_pointer(mask)?;
+
+                let thread_id = match pid {
+                    0 => this.active_thread(),
+                    _ => ThreadId::from(pid),
+                };
+
+                if cpusetsize != std::mem::size_of::<CpuAffinityMask>() as u64 {
```

Comment:
The man page says "smaller then", so shouldn't this be `<`?
If it's actually bigger you should probably initialize the remaining bytes to zero... though the manpage doesn't really say what happens.

We generally avoid `as` casts, please use `try_into().unwrap()`.